modules/gapi/src/streaming/onevpl/demux/async_mfp_demux_data_provider.cpp

```diff
@@ -0,0 +1,783 @@
+                            key << ", ready bitstream count: " <<
+                            bitstream_count);
+
+    return worker_buffer_count;
+}
+
+/////////////// IDataProvider methods ///////////////
+IDataProvider::mfx_codec_id_type MFPAsyncDemuxDataProvider::get_mfx_codec_id() const {
+    return codec;
+}
+
+mfxStatus MFPAsyncDemuxDataProvider::fetch_bitstream_data(std::shared_ptr<mfxBitstream> &out_bitsream) {
+    if (empty()) {
+        GAPI_LOG_DEBUG(nullptr, "[" << this << "] empty");
+        return MFX_ERR_MORE_DATA;
+    }
+    do {
+        if (out_bitsream) {
+            // make dmux buffer unlock for not empty bitstream
+            GAPI_LOG_DEBUG(nullptr, "[" << this << "] " <<
+                                    "bitstream before fetch: " <<
+                                    out_bitsream.get() <<
+                                    ", DataOffset: " <<
+                                    out_bitsream->DataOffset <<
+                                    ", DataLength: " <<
+                                    out_bitsream->DataLength);
+            if (out_bitsream->DataOffset < out_bitsream->DataLength) {
+                return MFX_ERR_NONE;
+            }
+
+            // cleanup
+            auto it = processing_key_to_buffer_mapping_storage.find(out_bitsream->Data);
+            if (it == processing_key_to_buffer_mapping_storage.end()) {
+                GAPI_LOG_WARNING(nullptr, "[" << this << "] " <<
+                                          "cannot find appropriate dmux buffer by key: " <<
+                                          static_cast<void*>(out_bitsream->Data));
+                GAPI_Assert(false && "invalid bitstream key");
+            }
+            if (it->second) {
+                it->second->Unlock();
+            }
+            processing_key_to_buffer_mapping_storage.erase(it);
+        }
+
+        // consume bitstream portion
+        if (processing_locked_buffer_storage.empty() &&
+            provider_state.load() == State::InProgress) {
+            // get worker data collected from another thread
+            consume_worker_data();
+        }
+
+        // EOF check: nothing to process at this point
+        if (processing_locked_buffer_storage.empty()) {
+            GAPI_DbgAssert(provider_state == State::Exhausted && "Source reader must be drained");
+            out_bitsream.reset();
+            return MFX_ERR_MORE_DATA;
+        }
+
+        out_bitsream = processing_locked_buffer_storage.front();
+        processing_locked_buffer_storage.pop();
+
+        GAPI_LOG_DEBUG(nullptr, "[" << this << "] "
+                                "bitstream after fetch: " <<
+                                out_bitsream.get() <<
+                                ", DataOffset: " <<
+                                out_bitsream->DataOffset <<
+                                ", DataLength: " <<
+                                out_bitsream->DataLength);
+    }
+    while (false);
```

Comment:
?

Reply:
thanks, it should be removed  after modifications